deploy.sh

```diff
@@ -87,7 +87,7 @@ if [[ ! -f $TAG_FILE ]]; then
 fi
 
 
-VERSION=$(head -n 1 $TAG_FILE)
+VERSION=$(head -n 1 $TAG_FILE | egrep '^[a-zA-Z0-9\_\.\-]+$')
```

Comment:
In this context, a tag is a git tag, and a version is a name of the software that is also a git tag. To the extent that a tag is a git ref, it's a git ref. I don't think it should allow forward slashes because we really want it to be a simple version tag. The version scheme has a human-friendly portion, an ISO8601 date of the git commit, e.g. 20230327, followed by a hyphen, followed by the first seven digits of the git commit. It is a compact but unambiguous version.

Other common version schemes might include an underscore or dot. I did not want this code to preclude the use of such version schemes, e.g. version 3.141 or 3.1.4.1 or 3_M or whatever. The main purpose is to provide a short list of characters that might be expected and valid. We don't want to include things like parens or dollar signs or pipes or spaces because those might be dangerous. So in this spirit, `..` is a "valid" version for the purposes of the regexp, but if it does not exist it will simply not be found at the URL (which is also true for `3.1` or `3-m`).

Reply:
Okay, thanks for explaining!

I still feel like I am missing enough context to properly evaluate this, and I think that rectifying that is not a good use of my time right now. Limiting the input is better than not doing so, so this is at least an increase in security - thank you for following up on that, @bickelj!

I'll add a few things that might be relevant, and you can choose whether to act on them or not. First, you might be intersted in `git-check-ref-format(1)`, with which you can validate a potential reference name before using it. Second - and see #80 for more detail - shellcheck reports that:

> - SC2086 (info): Double quote to prevent globbing and word splitting.
> - SC2196 (info): egrep is non-standard and deprecated. Use grep -E instead.

Reply:
I realize those would only take a few minutes to apply, but to test, etc., will take more than a few minutes. So I defer the linting to the linting issue because I think this change is an urgent security fix.